packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -1,23 +1,47 @@
 import type { LoadedStrapi } from '@strapi/types';
 import { omit, pick } from 'lodash/fp';
+
+import { scheduleJob } from 'node-schedule';
 import { HISTORY_VERSION_UID } from '../constants';
 
 import type { HistoryVersions } from '../../../../shared/contracts';
 
+const DEFAULT_RETENTION_DAYS = 90;
+
 const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
+  const state: {
+    deleteExpiredJob: ReturnType<typeof scheduleJob> | null;
+    isInitialized: boolean;
+  } = {
+    deleteExpiredJob: null,
+    isInitialized: false,
+  };
+
   /**
    * Use the query engine API, not the document service,
    * since we'll refactor history version to be just a model instead of a content type.
    * TODO: remove this comment once the refactor is done.
    */
   const query = strapi.db.query(HISTORY_VERSION_UID);
 
-  let isInitialized = false;
+  const getRetentionDays = (strapi: LoadedStrapi) => {
+    const licenseRetentionDays =
+      strapi.ee.features.get('cms-content-history')?.options.retentionDays;
+    const userRetentionDays: number = strapi.config.get('admin.history.retentionDays');
+
+    // Allow users to override the license retention days, but not to increase it
+    if (userRetentionDays && userRetentionDays < licenseRetentionDays) {
+      return userRetentionDays;
+    }
+
+    // User didn't provide retention days value, use the license or fallback to default
+    return licenseRetentionDays ?? DEFAULT_RETENTION_DAYS;
```

Comment:
I think this means that for EE customers they'll have the very big number as the default, which isn't what we want, instead we want them to have 90 by default. They should be able to have a very big number, but only if they explicitly set it themselves, not by default

I think this could make it work, without relying on the exact value of the big number:

```js
return Math.min(licenseRetentionDays, DEFAULT_RETENTION_DAYS);
```

Reply:
Oh right. I'm cool with your suggestion. But I'm also wondering if we should just return the default here. Is there ever a case where the license retention days is lower than the default? Otherwise we have the check above to use the required user config when retention days is a big number.

Reply:
I think we cheaper cloud plans could give you 7 days, in which we case we wouldn't want to grant the default 90 days

Reply:
Got it! updated with your suggestion :)